include/tins/tcp_ip/stream_identifier.h

```diff
@@ -30,6 +30,7 @@
 #ifndef TINS_TCP_IP_STREAM_ID_H
 #define TINS_TCP_IP_STREAM_ID_H
 
+#include "stream.h"
```

Comment:
You don't need this, you can just do a forward declaration of class `Stream` inside namespace `Tins::TCPIP` and then include this on the .cpp